test/index.spec.js

```diff
@@ -0,0 +1,93 @@
+/* eslint-env mocha */
+'use strict'
+
+const expect = require('chai').expect
+const IPFSFactory = require('ipfsd-ctl')
+const async = require('async')
+
+const DelegatedPeerRouting = require('../src')
+
+const getPeerMultiaddrs = (peer) => {
+  return peer.multiaddrs.toArray().map(ma => ma.toString())
+}
+
+describe('DelegatedPeerRouting', () => {
+  it('calls find peer on the connected node', function (done) {
+    this.timeout(1000 * 120)
+
+    const factory = IPFSFactory.create({ type: 'go' })
+    let ipfsd
+    let routing
+    let peer
+    async.waterfall([
+      (cb) => factory.spawn({
+        config: {
+          Addresses: {
+            API: '/ip4/127.0.0.1/tcp/0',
+            Swarm: [
+              '/ip4/0.0.0.0/tcp/0'
+            ]
+          }
+        }
+      }, cb),
+      (_ipfsd, cb) => {
+        ipfsd = _ipfsd
+        const opts = ipfsd.apiAddr.toOptions()
+        routing = new DelegatedPeerRouting({
+          protocol: 'http',
+          port: opts.port,
+          host: opts.host
+        })
+        ipfsd.api.swarm.peers(cb)
+      },
+      (peers, cb) => {
+        peer = peers[0]
+        const id = peer.peer.toB58String()
+        routing.findPeer(id, cb)
+      },
+      (newPeer, cb) => {
+        expect(newPeer.multiaddrs.size).to.be.above(0)
+        expect(getPeerMultiaddrs(newPeer)).to.contain(peer.addr.toString())
+        expect(newPeer.id).to.be.eql(peer.peer.toB58String())
+        cb()
+      }, (cb) => {
+        ipfsd.stop(cb)
+      }
+    ], done)
+  })
+
+  it('calls find peer on the connected node (using ipfs.io)', function (done) {
+    this.timeout(1000 * 120)
+    const routing = new DelegatedPeerRouting()
+    // Solus Bootstrapper Node ID
+    const id = 'QmSoLSafTMBsPKadTEgaXctDQVcqN88CNLHXMkTNwMKPnu'
```

Comment:
This is saturn which is in Singapore (= shitty connectivity), better use `QmSoLueR4xBeUbY9WZ9xGUUxunbKWcrNFTDAadQJmocnWm` (uranus) or `QmSoLV4Bbm51jM9C4gDYZQ9Cy3U6aXMJDAbzgu2fzaDs64` (venus) which are in NYC.

Sorry I didn't think of this when you asked earlier.

(Pro tip: the first character after QmSoL is the first letter of the hostname, i.e. s=saturn, v=venus, u=uranus)